metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala

```diff
@@ -0,0 +1,67 @@
+package scala.meta.internal.metals
+
+import java.nio.file.Path
+import MetalsEnrichments._
+import java.nio.file.Files
+import ch.epfl.scala.bsp4j.ScalacOptionsResult
+import scala.meta.internal.semanticdb.TextDocuments
+
+class SemanticdbIndexer(
+    referenceProvider: ReferenceProvider,
+    implementationProvider: ImplementationProvider
+) {
+
+  def onScalacOptions(scalacOptions: ScalacOptionsResult): Unit = {
+    for {
+      item <- scalacOptions.getItems.asScala
```

Comment:
```scala
scalacOptions.getItems.foreach { item =>
  val targetroot = item.targetroot
  onChangeDirectory(targetroot.resolve(Directories.semanticdb).toNIO)
}

Reply:
It's basically the same thing, I will leave it but switch {} to ()